rancher2/structure_agent_deployment_customization.go

```diff
@@ -0,0 +1,96 @@
+package rancher2
+
+import (
+	"encoding/json"
+	"fmt"
+
+	managementClient "github.com/rancher/rancher/pkg/client/generated/management/v3"
+)
+
+// Flatteners
+
+func flattenAgentDeploymentCustomization(in *managementClient.AgentDeploymentCustomization) []interface{} {
+	obj := make(map[string]interface{})
+	if in == nil {
+		return []interface{}{}
+	}
+
+	if len(in.AppendTolerations) > 0 {
+		obj["append_tolerations"] = in.AppendTolerations
+	}
+
+	if in.OverrideAffinity != nil {
+		obj["override_affinity"] = in.OverrideAffinity
+	}
+
+	if in.OverrideResourceRequirements != nil {
+		obj["override_resource_requirements"] = in.OverrideResourceRequirements
+	}
+
+	return []interface{}{obj}
+}
+
+// Expanders
+
+func expandAgentDeploymentCustomizationOverrideResourceRequirements(p []interface{}) *managementClient.ResourceRequirements {
+	obj := &managementClient.ResourceRequirements{}
+	if p == nil || len(p) == 0 || p[0] == nil {
+		return nil
+	}
+
+	for i := range p {
+		in := p[i].(map[string]interface{})
+
+		obj.Limits = make(map[string]string)
+		obj.Requests = make(map[string]string)
+
+		if v, ok := in["cpu_limit"].(string); ok && len(v) > 0 {
+			obj.Limits["cpu"] = v
+		}
+
+		if v, ok := in["cpu_request"].(string); ok && len(v) > 0 {
+			obj.Requests["cpu"] = v
```

Comment:
It's interesting..
The management client that TF rke1 uses when creating/updating clusters has a resource requirement type of

```
type ResourceRequirements struct {
    Limits   map[string]string `json:"limits,omitempty" yaml:"limits,omitempty"`
    Requests map[string]string `json:"requests,omitempty" yaml:"requests,omitempty"`
}
```

Keeping it as a map[string]string and thus, "cpu" instead of corev1.ResourceCPU works fine during testing. I use the corev1 types for v2 prov.

Reply:
`corev1.ResourceCPU` is a constant, not a type

Reply:
@jakefhyde They are type `ResourceName` and not type `string` so I'd have to do

```
obj.Limits[string(corev1.ResourceCPU)] = v
```

Which I don't like. I think we should keep them as strings in the non-v2 files.